engine/consensus/compliance/engine_test.go

```diff
@@ -187,35 +213,29 @@ func (cs *ComplianceSuite) TestSubmittingMultipleEntries() {
 		// store the data for retrieval
 		cs.headerDB[block.Header.ParentID] = cs.head
 		cs.hotstuff.On("SubmitProposal", block.Header, cs.head.View).Return()
-		_ = cs.engine.Process(network.ConsensusCommittee, originID, proposal)
+		err := cs.engine.Process(network.ConsensusCommittee, originID, proposal)
+		cs.Assert().NoError(err)
 		wg.Done()
 	}()
 
+	// wait for all messages to be delivered to the engine message queue
 	wg.Wait()
-
-	time.Sleep(time.Second)
+	// wait for the votes queue to drain
+	assert.Eventually(cs.T(), func() bool {
+		return cs.engine.pendingVotes.(*engine.FifoMessageStore).Len() == 0
+	}, time.Second, time.Millisecond*10)
+	// stop the engine
+	cs.cancel()
+	<-cs.engine.Done()
 
 	// check that submit vote was called with correct parameters
 	cs.hotstuff.AssertExpectations(cs.T())
 	cs.voteAggregator.AssertExpectations(cs.T())
 }
 
-// TestProcessUnsupportedMessageType tests that Process and ProcessLocal correctly handle a case where invalid message type
-// was submitted from network layer.
-func (cs *ComplianceSuite) TestProcessUnsupportedMessageType() {
-	invalidEvent := uint64(42)
-	err := cs.engine.Process("ch", unittest.IdentifierFixture(), invalidEvent)
-	// shouldn't result in error since byzantine inputs are expected
-	require.NoError(cs.T(), err)
-	// in case of local processing error cannot be consumed since all inputs are trusted
-	err = cs.engine.ProcessLocal(invalidEvent)
-	require.Error(cs.T(), err)
-	require.True(cs.T(), engine.IsIncompatibleInputTypeError(err))
-}
-
 // TestOnFinalizedBlock tests if finalized block gets processed when send through `Engine`.
 // Tests the whole processing pipeline.
-func (cs *ComplianceSuite) TestOnFinalizedBlock() {
+func (cs *EngineSuite) TestOnFinalizedBlock() {
```

Comment:
We should also fix this test, it uses `require.Eventually` with `AssertCalled`

Reply:
Nice catch 👍

Reply:
For my learning: is the old implementation (below) actually buggy or just unnecessarily verbose?
https://github.com/onflow/flow-go/blob/f7c9481e2d93a63a82160de645d2d4affc0b584e/engine/consensus/compliance/engine_test.go#L227-L230

Reply:
@AlexHentschel `AssertCalled` has  side-effects after first call that doesn't match assert it records a failure and reports it even if subsequent calls asserted to true.